homeassistant/auth/auth_store.py

```diff
@@ -286,13 +289,29 @@ async def async_get_refresh_token_by_token(
 
         return found
 
+    async def async_get_refresh_tokens(self) -> list[models.RefreshToken]:
+        """Get all refresh tokens."""
+        tokens: list[models.RefreshToken] = []
+        if self._users is None:
+            await self._async_load()
+            assert self._users is not None
+
+        for user in self._users.values():
+            tokens += list(user.refresh_tokens.values())
+
+        return tokens
+
     @callback
     def async_log_refresh_token_usage(
         self, refresh_token: models.RefreshToken, remote_ip: str | None = None
     ) -> None:
         """Update refresh token last used information."""
         refresh_token.last_used_at = dt_util.utcnow()
         refresh_token.last_used_ip = remote_ip
+        if refresh_token.expire_at:
+            refresh_token.expire_at = (
+                refresh_token.last_used_at + REFRESH_TOKEN_EXPIRATION
+            )
         self._async_schedule_save()
 
     async def _async_load(self) -> None:
```

Comment:
I wonder if this makes sense to load this after the registries in bootstrap and than we don't need all these coroutines and lock checks everywhere

Reply:
i don't know, but in my defense, this is existing code 😁 🙈

Reply:
Yeah it would be another PR for sure

We used to do the registries this way as well but changed it for similar reasons

Reply:
Probably will blow up in many ways

https://github.com/home-assistant/core/pull/108543

Reply:
Looks like it didn't blow up at all, and it does make the runtime checks, a bit faster and simpler